Updated colorpicker datatype to be usable as a style/setting on grid datatype

rasmusfjord opened the pull request:
If you use the colorpicker as a style or setting it dosnt work unless this fix is applied, because it works abit differently when its a style then as a standalone datatype.

Try with a setting/style like this:
[{
    "label": "Baggrundsfarve",
    "description": "Sæt en baggrundsfarve på rækken",
    "key": "background-color",
    "view": "colorpicker",
    "prevalues": [
      "#627991",
      "#a9c2d9",
      "#ccc9b0",
      "#8f5368",
      "#8f9f84",
      "#c8ccb5",
      "#3e424d",
      "#df791e",
      "#259ff4",
      "#bb0000"
    ]
  }
]
![image](https://user-images.githubusercontent.com/1286043/34941621-ce867b5c-f9f4-11e7-95b9-22a3f68a9c57.png)

bjarnef commented:
@rasmusfjord I don't see the "colorpicker" prevalue editor in your PR - only extend of the property editor.

Anyway I submitted a PR to add a color picker prevalue a long time ago, which both supported the old method with prevalues as comma delimited strings and objects with label and value.
https://github.com/umbraco/Umbraco-CMS/pull/1503

rasmusfjord commented:
@bjarnef Acutally the original colorpicker has been working as a prevalue editor, or been able to use as a style/setting since the beginning of 7. Ive been using it on an old 7.3.x which i just updated and noticed that it didnt work anymore. So I just adapted the controller so it could also work as a prevalue.

TL:DR
It worked before, broke somewhere beteween 7.3 - 7.7, and i fixed it :)

bjarnef commented:
@rasmusfjord okay, but then I don't think the Color Picker property editor has meant to be used as prevalue editor since it hasn't been available from here: https://github.com/umbraco/Umbraco-CMS/tree/dev-v7/src/Umbraco.Web.UI.Client/src/views/prevalueeditors

It should really exists in the same folders as the other ones, so you know which prevalue editors are available.

rasmusfjord commented:
That could be true, but since it does yeah it should probably be moved to give a better view of what is actually available. But would should be changed for that to work besides moving the view file? Is there a master package manifest somewhere which needs to be updated?

bjarnef commented:
If the changed from my PR is merged, it should also work with the setting/style config you specified :wink:

nul800sebastiaan commented:
Funnily enough I had just looked into #1503 the other day and discussed internally if we wanted to merge it. No promises but we wanted to merge that one @bjarnef - and indeed it will solve your problem too @rasmusfjord ! Thank you kindly for the work put in however #h5yr!
